pkg/intermediate/aggregate.go

```diff
@@ -148,37 +169,97 @@ func (a *AggregationProcess) ForAllRecordsDo(callback FlowKeyRecordMapCallBack)
 	return nil
 }
 
-// GetLastUpdatedTimeOfFlow provides the last updated time in the format of IPFIX
-// field "flowEndSeconds".
-func (a *AggregationProcess) GetLastUpdatedTimeOfFlow(flowKey FlowKey) (uint32, error) {
+func (a *AggregationProcess) deleteFlowKeyFromMap(flowKey util.FlowKey) error {
 	a.mutex.Lock()
 	defer a.mutex.Unlock()
-	record, exists := a.flowKeyRecordMap[flowKey]
+	return a.deleteFlowKeyFromMapWithoutLock(flowKey)
+}
+
+func (a *AggregationProcess) deleteFlowKeyFromMapWithoutLock(flowKey util.FlowKey) error {
+	_, exists := a.flowKeyRecordMap[flowKey]
 	if !exists {
-		return 0, fmt.Errorf("flow key is not present in the map")
+		return fmt.Errorf("flow key is not present in the map")
 	}
-	flowEndField, exists := record.Record.GetInfoElementWithValue("flowEndSeconds")
-	if exists {
-		return flowEndField.Value.(uint32), nil
-	} else {
-		return 0, fmt.Errorf("flowEndSeconds field is not present in the record")
+	delete(a.flowKeyRecordMap, flowKey)
+	return nil
+}
+
+// GetExpiryFromExpirePriorityQueue returns the earliest timestamp (active expiry
+// or inactive expiry) from expire priority queue.
+func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
```

Comment:
I wonder if we should have a minimum duration time here, to avoid always handling a single item per call
Maybe 100ms?

But I am not sure whether it's really important...

Reply:
Added this minimum duration here as we sometimes hit situation, where the top item's timestamp is already elapsed and returns a negative duration. The ticker in Antrea do not accept non-positive duration, so using a min duration here.

Reply:
Yes, that may not be guaranteed. I modified the logic a bit to cover that scenario. Ticker does not accept 0 too.